### PR TITLE
Enable streaming for PublishedArtifacts

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: pulpcore/tests/functional/api/using_plugin/test_content_app.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        populate_pulp(cls.cfg)
+        cls.contents = cls.client.get(FILE_CONTENT_PATH)['results'][:2]
+
+    def test_content_app_returns_404(self):
+        """Test that content app returns 404 on wrong url.
+
+        This test targets the following issue: 4278
+
+        * `Pulp Smash # <https://github.com/PulpQE/pulp-smash/issues/4278>`_
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/tests/functional/api/using_plugin/test_content_app.py]: Oh, copy'n'paste and not looking right...
Can i reference plan.io here?

[the remark this answers is not in the document]
Comment (dkliban) [reply pulpcore/tests/functional/api/using_plugin/test_content_app.py]: Yes

File: pulpcore/content/handler.py [hunk 2]
```diff
@@ -17,14 +17,19 @@
...
+class PathNotResolved(web_exceptions.HTTPNotFound):
     """
     The path could not be resolved to a published file.
 
     This could be caused by either the distribution, the publication,
     or the published file could not be found.
     """
-    pass
+
+    def __init__(self, *args, **kwargs):
```
Comment (gmbnomis) [inline pulpcore/content/handler.py]: I would prefer:

```
    def __init__(self, path, *args, **kwargs):
        """Initialize the Exception."""
        self.path = path
        super(PathNotResolved, self).__init__(*args, **kwargs)
```

File: pulpcore/tests/functional/api/using_plugin/test_content_app.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+        This test targets the following issue: 4278
+
+        * `<https://pulp.plan.io/issues/4278>`_
+
+        Do the following:
+
+        1. Create a repository that has at least one repository version.
+        2. Create a publisher.
+        3. Create a distribution and set the repository and publishera to the
```
Comment (gmbnomis) [inline pulpcore/tests/functional/api/using_plugin/test_content_app.py]: typo: publishera

File: pulpcore/content/handler.py [hunk 4]
```diff
@@ -17,14 +17,18 @@
...
 
     This could be caused by either the distribution, the publication,
     or the published file could not be found.
     """
-    pass
+
+    def __init__(self, path, *args, **kwargs):
+        """Initialize the Exception."""
+        self.path = path
+        super(PathNotResolved, self).__init__(*args, **kwargs)
```
Comment (dralley) [inline pulpcore/content/handler.py]: Python 3 doesn't need the (class, self) bit, just do super().__init__(...

**Comment by bmbouter** [comment]
Thank you!

**Pull request merged**